### ISSUE TITLE
Is the data numeric or character or formula?

### ISSUE THREAD

**Issue by jennybc** [jennybc/googlesheets#19]
The API doesn't do a great job communicating whether the data in a cell is numeric or character.  So far, in my rewrite, I've only taken an entire worksheet and I've used R native facilities for figuring this out, i.e. used `type.convert(…, as.is = TRUE)`. But I want to park some info here about how numeric vs. character data looks when I consume data from our Gapminder spreadsheet. In certain contexts it may be helpful to know this ….

```r
> devtools::load_all(".")
Loading gspreadr
> x <- "Gapminder"
> ss <- register(x)
Auto-refreshing stale OAuth token.
> str(ss)
              Spreadsheet title: Gapminder
     Date of gspreadr::register: 2015-02-07 17:09:38 PST
Date of last spreadsheet update: 2015-01-21 18:42:42 UTC

Contains 5 worksheets:
(Title): (Nominal worksheet extent as rows x columns)
Africa: 1000 x 26
Americas: 1000 x 26
Asia: 1000 x 26
Europe: 1000 x 26
Oceania: 1000 x 26

Key: 1hS762lIJd2TRUTVOqoOP7g-h4MDQs6b2vhkTzohg8bE
Worksheets feed: https://spreadsheets.google.com/feeds/worksheets/1hS762lIJd2TRUTVOqoOP7g-h4MDQs6b2vhkTzohg8bE/private/full
> americas <- get_ws(ss, ws = "Americas")
> GET_url <- americas$cellsfeed
> req <- gsheets_GET(GET_url)
> foo <- req$content %>% lfilt("entry")
> yo <- plyr::ldply(foo, function(x) {
+ dplyr::data_frame(content_text = x$content$text,
+ cell_text = x$cell$text,
+ cell_inputValue = x$cell$.attrs["inputValue"],
+ cell_numericValue = x$cell$.attrs["numericValue"])
+ }) %>% dplyr::select(-.id)
> identical(yo$content_text, yo$cell_text) # TRUE
[1] TRUE
> identical(yo$content_text, yo$cell_inputValue) # TRUE
[1] TRUE
> identical(yo$content_text, yo$cell_numericValue) # FALSE
[1] FALSE
```

At least on this spreadsheet, for the `entry` nodes of the cell feed, `content$text` is same as `cell$text` is same as `inputValue` attribute of `cell` for ALL cells. The `numericValue` attribute of `cell` is different for character-filled cells vs numeric-filled. Here's how the first 2 rows look -- the header row plus first row of data:

```r
> yo[1:12, c('cell_text', 'cell_numericValue')]
   cell_text cell_numericValue
1    country              <NA>
2  continent              <NA>
3       year              <NA>
4    lifeExp              <NA>
5        pop              <NA>
6  gdpPercap              <NA>
7  Argentina              <NA>
8   Americas              <NA>
9       2007            2007.0
10     75.32             75.32
11  40301927       4.0301927E7
12  12779.38          12779.38
```

**Comment by jennybc** [comment]
Leaving more informal work behind on what element-wise treatment with `type.convert()` will do.

```r
> lapply(yo$cell_text[1:12], type.convert, as.is = TRUE) %>% str
List of 12
 $ : chr "country"
 $ : chr "continent"
 $ : chr "year"
 $ : chr "lifeExp"
 $ : chr "pop"
 $ : chr "gdpPercap"
 $ : chr "Argentina"
 $ : chr "Americas"
 $ : int 2007
 $ : num 75.3
 $ : int 40301927
 $ : num 12779
> lapply(yo$cell_numericValue[1:12], type.convert, as.is = TRUE) %>% str
List of 12
 $ : logi NA
 $ : logi NA
 $ : logi NA
 $ : logi NA
 $ : logi NA
 $ : logi NA
 $ : logi NA
 $ : logi NA
 $ : num 2007
 $ : num 75.3
 $ : num 40301927
 $ : num 12779
```

Use on the `cell_text` actually seem preferable because it correctly distinguishes integer from numeric. But then I worry if there are numbers where scientific notation is necessary and `cell_numericValue` is a better input? I'm sure I will find out ...

**Comment by jennybc** [comment]
I anticipate some of the redundancy noted above will NOT be the case when cells contain formulas, which we've ignored thus far.

**Comment by jennybc** [comment]
Merged into #152. Closing.
